### PR TITLE
[Android] Fixed border inside frame not rendering

### LINKED ISSUE #18526: [regression/8.0.0] [Android] Border inside frame cause dont render content
### Description

Hello, when i put border inside a Frame -> Grid -> Border the content is empty dont render

**With border**
Screenshot_1699193855

Without border
Screenshot_1699193823

XML Code
```
 <VerticalStackLayout
  Spacing="25"
  Padding="30,0"
  VerticalOptions="Center">
     <Frame>
         <Grid RowDefinitions="auto, auto, auto, auto">
             <Border Stroke="#C49B33"
                     StrokeThickness="4"
                     StrokeShape="RoundRectangle 40,0,0,40"
                     Background="#2B0B98"
                     Padding="16,8"
                     HorizontalOptions="Center" Grid.Row="0">
                 <Label Text=".NET MAUI"
                        TextColor="White"
                        FontSize="18"
                        FontAttributes="Bold" />
             </Border>

             <Label Grid.Row="1"
                      Text="Login"
                      FontSize="18"
                      HorizontalOptions="Center" />

             <Button Grid.Row="2"
                      Text="Login"
                      Clicked="OnLogin"
                      HorizontalOptions="Center" />

             <Button Grid.Row="3"
                     Text="Settings"
                  Clicked="OnSettings"
                  HorizontalOptions="Center" />
         </Grid>
     </Frame>
 </VerticalStackLayout>
```

### Steps to Reproduce

1. Create MAUI .NET 8 Project
2. On page Add VerticalStackLayout, Frame, Grid, Border
3. The content is not displayed

### Link to public reproduction project repository

https://github.com/LaBoss/MAUITestes.git

### Version with bug

8.0.0-rc.2.9511

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.96

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11

### Did you find any workaround?

Dont use border

### Relevant log output

_No response_

### LINKED ISSUE #18755: Border will not render inside a Frame on Android
### Description

Border will not render inside a Frame on Android. Works OK on iOS and Catalyst. Not tested on Windows.

Microsoft.iOS - 17.0.8478
Microsoft.NETCore.App - 8.0.0
Microsoft.Extensions.Logging. Debug - 8.0.0
Microsoft.Maui.Controls - 8.0.3
Microsoft.Maui.Controls.Compatibility - 8.0.3

### Steps to Reproduce

1. Clone the attached repo
2. Build and run on Android Emulator

A Border with a Label should appear under the welcome message, but it doesn't.

### Link to public reproduction project repository

https://github.com/derlidio/BorderFrameBug.git

### Version with bug

Unknown/Other

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #20944: <Border /> property is not working while I write <Label /> property and passing a Text="", text is not being shown in the android emulator.
### Description

<ScrollView>
    <VerticalStackLayout>
        <Frame Margin="10" BackgroundColor="AliceBlue">
            <VerticalStackLayout>
                <Image Source="./Resources/Images/dotnet_bot.png" />
                <Label Text="Coding Droplets"
                            FontSize="20"
                            FontAttributes="Bold"
                            HorizontalOptions="Center"
                            TextColor="Sienna" />
            </VerticalStackLayout>
        </Frame>

        <Frame BackgroundColor="AntiqueWhite" Margin="10">
            <Border>
                <Label Text="Welcome" HorizontalOptions="Center" />
            </Border>
        </Frame>
    </VerticalStackLayout>
</ScrollView>

In this code, the text is not being shown under <Border /> property.

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.6 SR1

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.101

### Affected platforms

Android

### Affected platform versions

Android pixel 6 pro emulator API 32

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/18526
Fixes https://github.com/dotnet/maui/issues/18755
Fixes https://github.com/dotnet/maui/issues/20944

|Before|After|
|--|--|
|||

### CODE REVIEW

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs [hunk 1]
```diff
@@ -351,6 +351,18 @@ void UpdateContent()
 			if (content == null || _mauiContext == null)
 				return;
 
+			if (content is Border border)
+			{
+				foreach (var child in border.LogicalChildrenInternal)
+				{
+					if (child is IShape)
+					{
+						((VisualElement)child).IsVisible = false;
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:360]: Could you share more details about this change?

**Comment by jsuarezruiz** [comment]
Probably https://github.com/dotnet/maui/issues/20944 is also related with these changes.

File: src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs (hunk 1 above)
Comment (kubaflo) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:360]: @jsuarezruiz When we set the property `IsVisible` to `false` then the parent layout doesn't consider this shape when calculating the required size for its content.

https://github.com/kubaflo/maui/blob/9d50c370103718c654d67d51fc13827719f125fa/src/Controls/src/Core/Layout.cs#L663

I couldn't find a better fix for that. From what I understand `Frame` is rather a legacy component, so sometimes it has an unexpected behavior. I don't use it in my code, but many developers do and they often are not aware that it is better to use border to fix many issues. Therefore, I believe that this bug should be fixed to save them from being annoyed 😅

Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:360]: Can we just set `DisableLayout` to true?
https://github.com/dotnet/maui/blob/848202887c7a35d1250c4cb518e899ec65347545/src/Controls/src/Core/Layout.cs#L623

Comment (kubaflo) [reply src/Controls/src/Core/Compatibility/Handlers/Android/FrameRenderer.cs:360]: @PureWeen I tried but it doesn't work:
```
                       if (content is Border border)
			{
				var shape = border.LogicalChildrenInternal.FirstOrDefault(child => child is IShape);
				if (shape is VisualElement visualElement)
				{
					visualElement.DisableLayout = true; // doesn't work
					visualElement.IsVisible = false; // works
				}
			}
```

**Comment by PureWeen** [comment]
replaced by https://github.com/dotnet/maui/pull/23124

**Pull request closed without merging**